mobile/lib/pages/library/locked/locked.page.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:auto_route/auto_route.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/extensions/build_context_extensions.dart';
+import 'package:immich_mobile/providers/auth.provider.dart';
+import 'package:immich_mobile/providers/multiselect.provider.dart';
+import 'package:immich_mobile/providers/timeline.provider.dart';
+import 'package:immich_mobile/widgets/asset_grid/multiselect_grid.dart';
+
+@RoutePage()
+class LockedPage extends HookConsumerWidget {
+  const LockedPage({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    Widget infoBlock() {
+      return Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Center(
+          child: Text(
+            'no_locked_photos_message'.tr(),
+            style: context.textTheme.labelLarge,
+          ),
+        ),
+      );
+    }
+
+    return Scaffold(
+      appBar: ref.watch(multiselectProvider) ? null : const LockPageAppBar(),
+      body: MultiselectGrid(
+        renderListProvider: lockedTimelineProvider,
+        topWidget: infoBlock(),
+        editEnabled: false,
+        favoriteEnabled: false,
+        unfavorite: false,
+        archiveEnabled: false,
+        stackEnabled: false,
+        unarchive: false,
+      ),
+    );
+  }
+}
+
+class LockPageAppBar extends ConsumerWidget implements PreferredSizeWidget {
+  const LockPageAppBar({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return AppBar(
+      leading: IconButton(
+        onPressed: () {
+          ref.read(authProvider.notifier).lockPinCode();
+          context.maybePop();
+        },
```

Comment:
We can revert this to what it was previously, i.e,
```suggestion
        onPressed: () => context.maybePop(),
```
but use a `useEffect` hook inside the `LockedPage` widget to also handle the back gesture. Handling it in the app bar is useful only when the user navigates back using the AppBar leading icon. But most often than not, users usually use the back navigation button or the gesture to go back